### ISSUE TITLE
Refactor/Rethink Model outputs, wrt logits

### ISSUE THREAD

**Issue by zphang** [nyu-mll/jiant#1187]
(Documented as TODO in code.)

We abuse the name 'logits' in outputs for both `EmbeddingModel` and `RegressionModel`. Consider more outputs types, or rethinking our model output data structures.

Additional consideration: some model outputs have no loss.
